### DISCUSSION TITLE
Easy way to wrap an existing HttpRoute in Auth?

### DISCUSSION THREAD

**Discussion by Marcus-Rosti** [http4s/http4s#6870] [Q&A]
If I have a `val routes: HttpRoutes[F]` is there an easy way to turn it into `val authed: AuthedRoutes[Unit, F]`

I want to wrap some routes in a JWT and there's no user specific or ContextRequest needed. Just 👍 vs 👎 .

The other piece of context is that there are other routes, say, metrics & liveness, that don't need authentication. 

Maybe this is all just the nubile, not used to using HttpRoutes or Routers? Anyway, thanks!

**Comment by Marcus-Rosti** [comment]
@armanbilge thanks! Sorry about the issue!

**Comment by satorg** [comment]
@Marcus-Rosti do you mean some method like
```scala
def liftToAuthRoutes[F[_]](httpRoutes: HttpRoutes[F]): AuthedRoutes[Unit, F] =
  Kleisli { areq => httpRoutes(areq.req) }
```
❔

**Reply by Marcus-Rosti** [reply]
that's pretty similar to what I had
```
AuthedRoutes[Unit, F](r => MyHttpService(serviceImpl).scoreService.run(r.req))
```

but still has the same issue that whenever you pass something that isn't a valid route `<Root>/my/dumb/thing` it'll return 403  Unauth instead of 404 NotFound

**Comment by satorg** [comment]
Well, neither `HttpRoutes` nor `AuthedRoutes` do define what to return in a case when a request is not handled.
The fallback response has to be provided separately and explicitly. E.g. there's an extension syntax `.orNotFound` that can be added to any `HttpRoutes` instance to fallback to **404**. Therefore, if you'd like to enforce **404** as the fallback response for the inner `HttpRoutes`, then you could consider something like
```scala
def liftToAuthRoutes[F[_]: Functor](httpRoutes: HttpRoutes[F]): AuthedRoutes[Unit, F] =
  Kleisli { areq => OptionT.liftF(httpRoutes.orNotFound(areq.req)) }
```
